### PR TITLE
Redshift serverless endpoint

### LINKED ISSUE #24501: AWS Redshift Serverless support
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hi there,

I checked terraform documentation and don't see any way to create AWS Redshift Serverless endpoint. Looks like terraform supports only redshift clusters creation.

https://aws.amazon.com/redshift/redshift-serverless

### New or Affected Resource(s)

* [x] [`aws_redshiftserverless_namespace `](https://github.com/hashicorp/terraform-provider-aws/pull/25889)
* [x] [`aws_redshiftserverless_workgroup`](https://github.com/hashicorp/terraform-provider-aws/pull/26467)
* [ ] `aws_redshiftserverless_endpoint_access`

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #24501

Output from acceptance testing:

```
$ make testacc TESTS=TestAccRedshiftServerlessEndpointAccess_ PKG=redshiftserverless
--- PASS: TestAccRedshiftServerlessEndpointAccess_basic (883.76s)
--- PASS: TestAccRedshiftServerlessEndpointAccess_disappears_workgroup (552.98s)
--- PASS: TestAccRedshiftServerlessEndpointAccess_disappears (783.29s)
```

### CODE REVIEW

**Comment by DrFaust92** [comment]
Initial tests passing, tweaking sweepers/update tests

File: internal/service/redshiftserverless/wait.go [hunk 1]
```diff
@@ -87,3 +87,44 @@ func waitWorkgroupDeleted(conn *redshiftserverless.RedshiftServerless, name stri
 
 	return nil, err
 }
+
+func waitEndpointAccessActive(conn *redshiftserverless.RedshiftServerless, name string) (*redshiftserverless.EndpointAccess, error) {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{
+			"CREATING",
```
Comment (DrFaust92) [inline internal/service/redshiftserverless/wait.go:94]: trial and error from API

**Review by ewbankkit — COMMENTED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccRedshiftServerless' PKG=redshiftserverless ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/redshiftserverless/... -v -count 1 -parallel 3  -run=TestAccRedshiftServerless -timeout 180m
=== RUN   TestAccRedshiftServerlessEndpointAccess_basic
=== PAUSE TestAccRedshiftServerlessEndpointAccess_basic
=== RUN   TestAccRedshiftServerlessEndpointAccess_disappears_workgroup
=== PAUSE TestAccRedshiftServerlessEndpointAccess_disappears_workgroup
=== RUN   TestAccRedshiftServerlessEndpointAccess_disappears
=== PAUSE TestAccRedshiftServerlessEndpointAccess_disappears
=== RUN   TestAccRedshiftServerlessNamespace_basic
=== PAUSE TestAccRedshiftServerlessNamespace_basic
=== RUN   TestAccRedshiftServerlessNamespace_tags
=== PAUSE TestAccRedshiftServerlessNamespace_tags
=== RUN   TestAccRedshiftServerlessNamespace_disappears
=== PAUSE TestAccRedshiftServerlessNamespace_disappears
=== RUN   TestAccRedshiftServerlessWorkgroup_basic
=== PAUSE TestAccRedshiftServerlessWorkgroup_basic
=== RUN   TestAccRedshiftServerlessWorkgroup_tags
=== PAUSE TestAccRedshiftServerlessWorkgroup_tags
=== RUN   TestAccRedshiftServerlessWorkgroup_disappears
=== PAUSE TestAccRedshiftServerlessWorkgroup_disappears
=== CONT  TestAccRedshiftServerlessEndpointAccess_basic
=== CONT  TestAccRedshiftServerlessNamespace_disappears
=== CONT  TestAccRedshiftServerlessNamespace_basic
--- PASS: TestAccRedshiftServerlessNamespace_disappears (17.73s)
=== CONT  TestAccRedshiftServerlessNamespace_tags
--- PASS: TestAccRedshiftServerlessNamespace_basic (38.19s)
=== CONT  TestAccRedshiftServerlessEndpointAccess_disappears
--- PASS: TestAccRedshiftServerlessNamespace_tags (42.62s)
--- PASS: TestAccRedshiftServerlessWorkgroup_tags (528.60s)
=== CONT  TestAccRedshiftServerlessWorkgroup_disappears
--- PASS: TestAccRedshiftServerlessEndpointAccess_disappears (729.37s)
=== CONT  TestAccRedshiftServerlessWorkgroup_basic
--- PASS: TestAccRedshiftServerlessWorkgroup_basic (439.81s)
=== CONT  TestAccRedshiftServerlessEndpointAccess_disappears_workgroup
--- PASS: TestAccRedshiftServerlessWorkgroup_disappears (638.76s)
--- PASS: TestAccRedshiftServerlessEndpointAccess_basic (1173.96s)
--- PASS: TestAccRedshiftServerlessEndpointAccess_disappears_workgroup (624.71s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/redshiftserverless	1761.773s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**